packages/openapi-generator/src/parser/schema.ts

```diff
@@ -10,3 +22,124 @@ export function isArraySchemaObject(
 ): obj is OpenAPIV3.ArraySchemaObject {
   return obj?.type === 'array';
 }
+
+export function parseSchema(
```

Comment:
General concern: References. In the spec it is stated that references are simple in a sense that they do not extend an schema if there are `sibling` properties. This makes it simpleer for us, puh.

https://swagger.io/docs/specification/using-ref/

However, we should add tests for $ref cases which could be chained. This is allowed I think.

Reply:
As we never resolve schema references, this should not be a concern. Discussed.